tests/console/consoletest_setup.pm

```diff
@@ -21,7 +21,41 @@ sub run() {
 
     if (!check_var('ARCH', 's390x')) {
         # verify there is a text console on tty1
-        send_key_until_needlematch "tty1-selected", "ctrl-alt-f1", 6, 5;
+        for (1 .. 6) {
+            send_key 'ctrl-alt-f1';
+            if (check_screen("tty1-selected", 5)) {
+                last;
+            }
+        }
+        if (!check_screen "tty1-selected", 5) {    #workaround for bsc#977007
+            record_soft_failure "unable to switch to the text mode";
+            send_key 'ctrl-alt-backspace';         #kill X and log in again
+            send_key 'ctrl-alt-backspace';
+            if (get_var("NOAUTOLOGIN")) {          #copy from installation/first_boot.pm
```

Comment:
I meant if a NOAUTOLOGIN=0 job hits this bug, it will not go through those code copied from first_boot.pm, eg. https://openqa.opensuse.org/tests/173528 openSUSE GNOME test is auto-login enabled, if the bug also affects openSUSE, the "re-login" part on this PR will not work in that case.